crates/oxc_linter/src/rules/eslint/no_plusplus.rs

```diff
@@ -0,0 +1,215 @@
+
+    if operator == "++" {
+        return diagnostic.with_help("Use the assignment operator `+=` instead.");
+    } else if operator == "--" {
+        return diagnostic.with_help("Use the assignment operator `-=` instead.");
+    }
+
+    diagnostic
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoPlusplus {
+    /// Whether to allow `++` and `--` in for loop afterthoughts.
+    allow_for_loop_afterthoughts: bool,
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallow the unary operators `++`` and `--`.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Because the unary `++` and `--` operators are subject to automatic semicolon insertion, differences in whitespace
+    /// can change the semantics of source code. For example, these two code blocks are not equivalent:
+    ///
+    /// ```js
+    /// var i = 10;
+    /// var j = 20;
+    ///
+    /// i ++
+    /// j
+    /// // => i = 11, j = 20
+    /// ```
+    ///
+    /// ```js
+    /// var i = 10;
+    /// var j = 20;
+    ///
+    /// i
+    /// ++
+    /// j
+    /// // => i = 10, j = 21
+    /// ```
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// var x = 0; x++;
+    /// var y = 0; y--;
+    /// for (i = 0; i < l; i++) {
+    ///     doSomething(i);
+    /// }
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// var x = 0; x += 1;
+    /// var y = 0; y -= 1;
+    /// for (i = 0; i < l; i += 1) {
+    ///    doSomething(i);
+    /// }
+    /// ```
+    NoPlusplus,
+    restriction,
+);
+
+impl Rule for NoPlusplus {
+    fn from_configuration(value: serde_json::Value) -> Self {
+        let obj = value.get(0);
+        Self {
+            allow_for_loop_afterthoughts: obj
+                .and_then(|v| v.get("allowForLoopAfterthoughts"))
+                .and_then(serde_json::Value::as_bool)
+                .unwrap_or(false),
+        }
+    }
+
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::UpdateExpression(expr) = node.kind() else {
+            return;
+        };
+
+        if self.allow_for_loop_afterthoughts && is_for_loop_afterthought(node, ctx) {
+            return;
+        }
+
+        ctx.diagnostic(no_plusplus_diagnostic(expr.span, expr.operator.as_str()));
+    }
+}
+
+/// Whether the given AST node is a ++ or -- inside of a for-loop update.
+fn is_for_statement_update(node: &AstNode, ctx: &LintContext) -> bool {
+    let Some(parent) = ctx.nodes().parent_node(node.id()) else {
+        return false;
+    };
+    let AstKind::ForStatement(for_stmt) = parent.kind() else {
+        return false;
+    };
+
+    for_stmt.update.as_ref().is_some_and(|update| is_eq_node_expr(node, update))
+}
+
+/// Checks if the given node is equivalent to the given expression (i.e., they have the same span).
+fn is_eq_node_expr(node: &AstNode, expr: &Expression) -> bool {
```

Comment:
This part feels bad, not sure if there's a better way to do this. I noticed that we had similar code here:https://github.com/oxc-project/oxc/blob/16009c09fdb22fc89c6c23cef164635bf13f6723/crates/oxc_linter/src/rules/eslint/no_await_in_loop.rs#L104 so it might be worth making this a more general method.